data_struct/odd_pos_replace.py

```diff
@@ -0,0 +1,9 @@
+print "***Replacing character on odd position with X on string***"
+string = raw_input("\nEnter the string: ")
+for i in range(len(string)): 
```

Comment:
Change the code using enumerate.
Did you verified this code for all inputs.
```
In [3]: "teset".replace("teset"[1],"X")
Out[3]: 'tXsXt'

In [4]: "teset".replace("teset"[0],"X")
Out[4]: 'XeseX'
 
```
i.e replace actually change all the occurrence of the old string (first argument)

I think you need to change your logic.